### PR TITLE
Fix comment utilities in C++ writer

### PR DESCRIPTION
**Pull request by tiffany1618**
- Fix `addSeparatedPreComment` utility in `CppWriterUtils` for multiline annotations
- Fix `writeEnumConstant` utility in `CppWriterUtils`  for multiline annotations and fix Doxygen comment type
- Add test case for multiline annotations

### CODE REVIEW

File: compiler/tools/fpp-to-cpp/test/component/ActiveCommandsComponentAc.ref.hpp [hunk 1]
```diff
@@ -95,23 +95,23 @@ class ActiveCommandsComponentBase :
...
     //! Command opcodes
     enum {
-      OPCODE_CMD_SYNC = 0x0, //! A sync command with no params
-      OPCODE_CMD_SYNC_PRIMITIVE = 0x10, //! An async command with primitive params
-      OPCODE_CMD_SYNC_STRING = 0x11, //! A sync command with string params
-      OPCODE_CMD_SYNC_ENUM = 0x12, //! A sync command with enum params
-      OPCODE_CMD_SYNC_ARRAY = 0x13, //! A sync command with array params
-      OPCODE_CMD_SYNC_STRUCT = 0x14, //! A sync command with struct params
-      OPCODE_CMD_GUARDED = 0x15, //! A guarded command with no params
-      OPCODE_CMD_GUARDED_PRIMITIVE = 0x16, //! An aguarded command with primitive params
```
Comment (bocchino) [inline compiler/tools/fpp-to-cpp/test/component/ActiveCommandsComponentAc.ref.hpp:105]: Typo: aguarded --> guarded

**Review by bocchino — COMMENTED** [review]
Looks good! Just one comment.

File: compiler/tools/fpp-to-cpp/test/component/ActiveCommandsComponentAc.ref.hpp (hunk 1 above)
Comment (tiffany1618) [reply compiler/tools/fpp-to-cpp/test/component/ActiveCommandsComponentAc.ref.hpp:105]: Thanks! Fixed it.

**Review by bocchino — APPROVED** [review]
Looks good!

**Pull request merged**